### ISSUE TITLE
ES6 Destruct on Catch

### ISSUE THREAD

**Issue by jupl** [eslint/eslint#2123]
Given the following snippet, with ESLint I get an error on the `catch` line with `Unexpected token {`

```js
function x({a}) {
  try {
    const {b} = a;
  }
  catch({stack}) {
  }
}
```

**Comment by nzakas** [comment]
This is actually a parser error, can you file an issue with Espree (https://github.com/eslint/espree/issues)?

**Comment by nzakas** [comment]
Actually, can verify that you've enabled destructuring in your config?

**Comment by jupl** [comment]
Can do. Destructuring is on since the other destructuring in the snippet does not result in errors.

**Comment by nzakas** [comment]
The parser has been updated. You should be able to get this latest update by reinstalling eslint `npm i eslint`.

**Comment by jupl** [comment]
Thank you, fix did work. I did get a side effect that I'll report as another issue.
